Reword External Function Calls paragraph in docs to improve clarity

istareatscreens opened the pull request:
Changed:
![image](https://user-images.githubusercontent.com/46049787/133007349-385533ef-82f0-435e-ac8e-5f183f915294.png)
To:
![image](https://user-images.githubusercontent.com/46049787/133007286-f93872d5-3e5c-4c8b-8b0e-197780a4a5f5.png)

cameel commented on docs/control-structures.rst:
```diff
@@ -65,11 +65,11 @@ uses up at least one stack slot and there are only 1024 slots available.
-Please note that function calls on ``this`` cannot be used in the constructor,
-as the actual contract has not been created yet.
+Functions can also be called using ``this.g(8);`` and ``c.g(2);`` notation, where 
+c is a contract instance and g is a function belonging to c. Calling 
```
```suggestion
``c`` is a contract instance and ``g`` is a function belonging to ``c``. Calling 
```

cameel commented on docs/control-structures.rst:
```diff
@@ -65,11 +65,11 @@ uses up at least one stack slot and there are only 1024 slots available.
-will be called "externally", via a message call and not directly via jumps.
-Please note that function calls on ``this`` cannot be used in the constructor,
-as the actual contract has not been created yet.
+Functions can also be called using ``this.g(8);`` and ``c.g(2);`` notation, where 
+c is a contract instance and g is a function belonging to c. Calling 
+function g via either method results in it being called "externally", via a message
+call and not directly via jumps. Please note that function calls on ``this`` cannot
+be used in the constructor, as the actual contract has not been created yet.
```
Starting a new sentence always on a new line causes less rewrapping and gives better diffs (it would for example make it easier to see that you did not modify the last sentence):
```suggestion
Functions can also be called using ``this.g(8);`` and ``c.g(2);`` notation, where 
c is a contract instance and g is a function belonging to c.
Calling function g via either method results in it being called "externally", via a
message call and not directly via jumps.
Please note that function calls on ``this`` cannot be used in the constructor,
as the actual contract has not been created yet.
```

cameel reviewed (commented):
Thanks! Looks like an improvement. There are just two small formatting issues.

cameel commented:
Looks like CI is failing because you left some trailing whitespace in the paragraph.

Also, please squash your commits so that each commit is one is a single, consistent and self-contained change, without any going back and forth.

chriseth commented on docs/control-structures.rst:
```diff
@@ -65,11 +65,11 @@ uses up at least one stack slot and there are only 1024 slots available.
-as the actual contract has not been created yet.
+Functions can also be called using ``this.g(8);`` and ``c.g(2);`` notation, where 
+c is a contract instance and g is a function belonging to c. Calling 
+function g via either method results in it being called "externally", via a message
```
Maybe we should avoid the word 'method' in this context (sometimes functions are called methods) and just use "way" instead.

istareatscreens commented:
Hey sorry, I am a bit new to this so I squashed the commits and added the suggested changes. Please let me know if there is anything else you want me to do.

chriseth commented on docs/control-structures.rst:
```diff
@@ -65,9 +65,10 @@ uses up at least one stack slot and there are only 1024 slots available.
-The expressions ``this.g(8);`` and ``c.g(2);`` (where ``c`` is a contract
-instance) are also valid function calls, but this time, the function
-will be called "externally", via a message call and not directly via jumps.
+Functions can also be called using ``this.g(8);`` and ``c.g(2);`` notation, where
+``c`` is a contract instance and ``g`` is a function belonging to ``c``.
+Calling function g via either way results in it being called "externally", via a
+message call and not directly via jumps.
```
```suggestion
Functions can also be called using the ``this.g(8);`` and ``c.g(2);`` notation, where
``c`` is a contract instance and ``g`` is a function belonging to ``c``.
Calling the function ``g`` via either way results in it being called "externally", using a
message call and not directly via jumps.
```

cameel commented:
> Hey sorry, I am a bit new to this so I squashed the commits and added the suggested changes. Please let me know if there is anything else you want me to do.

No worries! The commit looks good apart from its description - looks like the descriptions from the squashed commits got in there.

istareatscreens committed:
Reword External Function Calls paragraph in docs

Reword External Function Calls opening paragraph to improve clarity

istareatscreens commented:
Fixed the commit description as requested let me know if you need me to change anything else

cameel commented:
There's still one unaddressed review comment: https://github.com/ethereum/solidity/pull/11937#discussion_r708082365 (added today).

istareatscreens commented:
Definitely missed it, sorry about that. I added the suggested changes.

chriseth commented:
Thanks a lot!
